precompile/modules/minitia_stdlib/sources/vip/score.move

```diff
@@ -290,22 +318,36 @@ module minitia_std::vip_score {
         deployer: &signer,
         stage: u64,
         addrs: vector<address>,
-        scores: vector<u64>
+        update_scores: vector<u64>
     ) acquires ModuleStore {
+
         assert!(
-            vector::length(&addrs) == vector::length(&scores),
+            vector::length(&addrs) == vector::length(&update_scores),
             error::invalid_argument(ENOT_MATCH_LENGTH)
         );
+
         prepare_stage(deployer, stage);
 
+        let module_store = borrow_global_mut<ModuleStore>(@minitia_std);
+
+        assert!(
+            table::contains(&module_store.scores, stage),
+            error::invalid_argument(EINVALID_STAGE)
+        );
+
+        let scores = table::borrow_mut(&mut module_store.scores, stage);
+        assert!(
+            !scores.is_finalized,
+            error::invalid_argument(EFINALIED_STAGE)
+        );
         vector::enumerate_ref(
             &addrs,
             |i, addr| {
-                update_score(
-                    deployer,
+                update_score_internal(
```

Comment:
if we want to use update_score_internal directly we need to add permission check.

Reply:
perhaps if you are referring to the permission check for the deployer, it is within the prepare_stage function.

Reply:
okay, i got it

Reply:
then i'll add failed test case of non-permissioned deployer

Reply:
good